pkg/konnector/konnector_reconcile.go

```diff
@@ -120,6 +120,16 @@ func (r *reconciler) reconcile(ctx context.Context, binding *kubebindv1alpha1.AP
 	ctrl, err := r.newClusterController(
 		binding.Spec.KubeconfigSecretRef.Namespace+"/"+binding.Spec.KubeconfigSecretRef.Name,
 		providerNamespace,
+		func(svcBinding *kubebindv1alpha1.APIServiceBinding) bool {
+			r.lock.RLock()
+			defer r.lock.RUnlock()
+			// for the very first time, ctrlContext is nil
+			// and need to check for this as well
+			if binding.Name == svcBinding.Name {
+				return true
+			}
+			return ctrlContext != nil && ctrlContext.serviceBindings.Has(svcBinding.Name)
```

Comment:
This won't work here. You have to move the whole block of line 140-145 above this closure.